tests/Doctrine/Tests/ORM/UnitOfWorkTest.php

```diff
@@ -322,6 +323,34 @@ public function testRemovedAndRePersistedEntitiesAreInTheIdentityMapAndAreNotGar
     }
 
     /**
+     * @group 5579
+     */
+    public function testEntityChangeSetNotClearAfterFlushOnEntityOrArrayOfEntity()
+    {
+        // Create and Set first entity
+        $entity1 = new NotifyChangedEntity;
+        $entity1->setData('thedata');
+        $this->_unitOfWork->persist($entity1);
+
+        // Create and Set second entity
+        $entity2 = new NotifyChangedEntity;
+        $entity2->setData('thedata');
+        $this->_unitOfWork->persist($entity2);
+
+        $this->_unitOfWork->commit($entity1);
+        $this->assertCount(1, $this->_unitOfWork->getEntityChangeSet($entity2));
+
+        // Create and Set third entity
+        $entity3 = new NotifyChangedEntity;
+        $entity3->setData('thedata');
+        $this->_unitOfWork->persist($entity3);
+
+        $this->_unitOfWork->commit([$entity1,$entity2]);
+        $this->assertCount(1, $this->_unitOfWork->getEntityChangeSet($entity3));
```

Comment:
Should probably check that there's no changeset for `$entity1` and `$entity2` here.